tests/unittests/CoreGraphics/CGImageTests.mm

```diff
@@ -0,0 +1,139 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <TestFramework.h>
+#import <Starboard.h>
+#import <CoreGraphics/CGImage.h>
+#import <Foundation\Foundation.h>
+#import <Starboard/SmartTypes.h>
+#import "TestUtils.h"
+
+typedef struct {
+    NSString* fileName;
+    size_t height;
+    size_t width;
+    bool isMask;
+} _ImageInfo;
+
+static const int number_of_jpeg_files = 4;
+static const _ImageInfo imagesJPEG[number_of_jpeg_files] = {
+    { @"dog.jpg", 639, 960, false },
+    { @"CroppedPhoto2.jpg", 200, 200, false },
+    { @"bowl.jpg", 4000, 6000, false },
+    { @"photo.jpg", 1024, 683, false },
+};
+
+static const int number_of_png_files = 2;
+static const _ImageInfo imagesPNG[number_of_png_files] = {
+    { @"dog.PNG", 700, 1044, false }, { @"microsoft.png", 136, 370, false },
+};
+
+#define EXPECT_IMAGE_DATA(image, imageInfo)               \
+    EXPECT_TRUE(image != NULL);                           \
+    EXPECT_EQ(imageInfo.isMask, CGImageIsMask(image));    \
+    EXPECT_EQ(imageInfo.height, CGImageGetHeight(image)); \
+    EXPECT_EQ(imageInfo.width, CGImageGetWidth(image));
+
+static CGImageRef createJPEG(NSString* path) {
+    CFDataRef data = (CFDataRef)[NSData dataWithContentsOfFile:getPathToFile(path)];
+    CGDataProviderRef dataProvider = CGDataProviderCreateWithCFData(data);
+    CGImageRef ret = CGImageCreateWithJPEGDataProvider(dataProvider, NULL, NO, kCGRenderingIntentDefault);
+    CGDataProviderRelease(dataProvider);
+    return ret;
+}
+
+static CGImageRef createPNG(NSString* path) {
+    CFDataRef data = (CFDataRef)[NSData dataWithContentsOfFile:getPathToFile(path)];
```

Comment:
It's autoreleased.

---
In reply to: [84551298](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84551298) [](ancestors = 84551298)